src/Components/WebAssembly/WebAssembly/test/Hosting/WebAssemblyHostConfigurationTest.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Xunit;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Memory;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Hosting
+{
+    public class WebAssemblyHostConfigurationTest
+    {
+        [Fact]
+        public void CanSetAndGetConfigurationValue()
+        {
+            // Arrange
+            var initialData = new Dictionary<string, string>() { { "color", "blue" } };
+            var memoryConfig = new MemoryConfigurationSource { InitialData = initialData };
+            var configuration = new WebAssemblyHostConfiguration();
+
+            // Act
+            configuration.Add(memoryConfig);
+            configuration["type"] = "car";
+
+            // Assert
+            Assert.Equal("car", configuration["type"]);
+            Assert.Equal("blue", configuration["color"]);
+        }
+
+        [Fact]
+        public void SettingValueUpdatesAllProviders()
+        {
+            // Arrange
+            var initialData = new Dictionary<string, string>() { { "color", "blue" } };
+            var source1 = new MemoryConfigurationSource { InitialData = initialData };
+            var source2 = new MemoryConfigurationSource { InitialData = initialData };
+            var configuration = new WebAssemblyHostConfiguration();
+
+            // Act
+            configuration.Add(source1);
+            configuration.Add(source2);
+            configuration["type"] = "car";
+
+            // Assert
+            Assert.Equal("car", configuration["type"]);
+        }
+
+        [Fact]
+        public void CanGetChildren()
+        {
+            // Arrange
+            var initialData = new Dictionary<string, string>() { { "color", "blue" } };
+            var memoryConfig = new MemoryConfigurationSource { InitialData = initialData };
+            var configuration = new WebAssemblyHostConfiguration();
+
+            // Act
+            configuration.Add(memoryConfig);
+            var children = configuration.GetChildren();
+
+            // Assert
+            Assert.NotNull(children);
+            Assert.NotEmpty(children);
+        }
```

Comment:
Should we have a test with two or more providers here to ensure that the results are what we expect? Using only one memory provider only covers the simplest case.

Could I suggest we also test a config that is a bit more complex? (like with nested sections) to ensure that the expected override behavior is preserved.

Reply:
I copied a few more of the nested test cases from [this](https://github.com/dotnet/runtime/blob/9e608fcecbabbc651d21582416e5807038ca7142/src/libraries/Microsoft.Extensions.Configuration/tests/ConfigurationTest.cs).

I added a test with an in-memory provider and a custom one. The E2E tests validate with a JsonStreamProvider and a MemoryProvider.